src/games/gameLogic.ts

```diff
@@ -1,64 +1,53 @@
-const board1 = [
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0']
-]
+import { ValidatorConstraint, ValidatorConstraintInterface } from 'class-validator'
+import { Board, emptyBoard, Row} from './entities';
 
-const board2 = [
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0']
-]
+@ValidatorConstraint()
+export class IsBoard implements ValidatorConstraintInterface {
+
+  validate(board: Board) {
+    return board.length === 10 &&
+      board.every(row =>
+        row.length === 10 &&
+        row.every(value => typeof(value)==='string')
+      )
+  }
+}
 
 // start a game
 // set maximium number of squares per ship here as well? (= define ships)
 export const newGame = () => {
-  return [board1, board2]
+  return emptyBoard
 }
 
 // give a square a hit value to be able to change class for css styling
-export const hit = (board, x, y) => {
+export const hit = (board: Board, x: number, y: number) => {
   board[x][y] += 'x'
```

Comment:
ohhh this looks so professional! Me likey (do I win a price for stupid comments? XD )

Reply:
:1st_place_medal: :stuck_out_tongue_winking_eye: